shoop/admin/modules/products/views/edit.py

```diff
@@ -104,11 +105,56 @@ def form_valid(self, form):
             form.forms["attributes"].save()
 
 
+class BaseProductMediaFormPart(FormPart):
+    def get_form_defs(self):
+        yield TemplatedFormDef(
+            self.name,
+            self.formset,
+            template_name="shoop/admin/products/_edit_media_form.jinja",
+            required=False,
+            kwargs={"product": self.object, "languages": settings.LANGUAGES}
+        )
+
+
+class ProductMediaFormPart(BaseProductMediaFormPart):
+    name = "media"
+    priority = -700
+    formset = ProductMediaFormSet
+
+    def form_valid(self, form):
+        if self.name in form.forms:
+            frm = form.forms[self.name]
+            instances = frm.save()
+            for inst in instances:
+                inst.kind = ProductMediaKind.GENERIC_FILE
+                inst.save()
+
+
+class ProductImageMediaFormPart(BaseProductMediaFormPart):
+    name = "images"
+    priority = -1001
+    formset = ProductImageMediaFormSet
+
+    def form_valid(self, form):
+        if self.name in form.forms:
+            frm = form.forms[self.name]
+            frm.save()
+
+            # delete possible link
```

Comment:
For some reason, this (`on_delete`) didn't work when I first prototyped the feature. It does now, so I will remove these lines.